File: corona.java



import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

class corona {

	static boolean report(String tmp1, String tmp2,
								int m, int n)
	
	{
		int j = 0;
		for (int i = 0; i < n && j < m; i++)
		{
			if (tmp1.charAt(j) == tmp2.charAt(i))
			{
				j++;

			}
	}
		return (j == m);
	}

	public static void main(String[] args)throws IOException
	{
		InputStreamReader read=new InputStreamReader(System.in);
		BufferedReader in=new BufferedReader(read);
		System.out.println("Enter Virus Name");
		String str1=in.readLine();
		str1=str1.toLowerCase(); //convert String into Lower Case
		System.out.println("The name of the Virus is "+str1);
		System.out.println("Enter number of people");
		int n,i;
		
		String str2;
		n=Integer.parseInt(in.readLine()); //input no of people
		String arr[]=new String[n]; //Declare String of array with size= no of people
			for(i=0;i<n;i++) {
				System.out.println("Enter the Sample report of person "+(i+1));
				str2=in.readLine();
				str2=str2.toLowerCase();//Convert String in lower case
				arr[i]=str2;
			}
			for( i=0;i<arr.length;i++) {
				String tmp1 = arr[i];
		        int length1 = tmp1.length();
		        int length2 = str1.length();
		        boolean result = report(tmp1, str1, length1, length2);
		 
		        if (result)
				System.out.println("POSITIVE");
			else
				System.out.println("NEGATIVE");
			}

		}
		
	}

File: corona.js
const prompt = require("prompt-sync")(); //Run npm install prompt-sync in the terminal

const str1 = prompt("Enter the virus name ");
str1=str1.toLowerCase(); //convert the sample virus into Lowercase
console.log(`The name of the virus is ${str1}`);

const n = prompt("Enter number of people between 1 to 10");// input no of people
console.log(`The number of people is ${n}`);

const arr = [];
for (let i = 0; i < n; i++) {
  arr[i] = prompt("Enter the Sample report of person " + (i + 1)+" ");
  arr[i]=arr[i].toLowerCase(); //convert string to lower case
}
console.log(arr);
const len = arr.length; //calculate the length of arr
const length1 = str1.length;
let str2;
for (i = 0; i < len; i++) {
  str2 = arr[i];
  const length2 = str2.length;
  let result = report(str2, str1, length2, length1);// function call 
  if (result) console.log("POSITIVE");//receive boolean value from function report
  else console.log("NEGATIVE");
}
function report(tmp1, tmp2, m, n) {
  let j = 0;
  for (let i = 0; i < n && j < m; i++) {
    if (tmp1[j] == tmp2[i]) { //if condition to check whether the character of tmp1 exists in tmp2
      j++; //increment to the next element in the tmp1
    }
  }
  return j == m; //return boolean value
}
